File: src/main/java/edu/kit/iti/formal/stvs/model/table/StringEditable.java
package edu.kit.iti.formal.stvs.model.table;

import javafx.beans.property.StringProperty;

/**
 * This interface is implemented by all classes with a string representation that can be read and
 * changed.
 *
 * @author Benjamin Alt
 */
public interface StringEditable extends StringReadable {

  String getAsString();

  void setFromString(String input);

  StringProperty stringRepresentationProperty();
}

File: stvs/src/main/kotlin/edu/kit/iti/formal/stvs/model/table/StringEditable.kt
package edu.kit.iti.formal.stvs.model.table

import javafx.beans.property.StringProperty

/**
 * This interface is implemented by all classes with a string representation that can be read and
 * changed.
 *
 * @author Benjamin Alt
 */
interface StringEditable : StringReadable {
    override val stringRepresentationProperty: StringProperty
    fun setFromString(input: String) = stringRepresentationProperty.set(input)
}
